### ISSUE TITLE
Set consistency level for cassandra using kundera

### ISSUE THREAD

**Issue by ansarrafique** [Impetus/kundera#557]
Is there any way to set the consistency level for cassandra using kundera ? The idea option is to set at entity level or a session level for each query execution. I am using kundera-cassandra 2.9.

**Comment by mevivs** [comment]
Please refer https://github.com/impetus-opensource/Kundera/issues/331

**Comment by ansarrafique** [comment]
Can I also specify the consistency level for entity manager like

@PersistenceContext(type = PersistenceContextType.EXTENDED, properties = {@PersistenceProperty(name="consistency.level", value="QUORUM")})  
Does it work ? 

Also, is there any way to set consistency level for entitymanager rather than directly getting thrift client then I have to execute queries using thrift client not entity manager rite ?

What is the default consistency level for cassandra using kundera ? Same as default cassandra which is ONE ?

**Comment by mevivs** [comment]
```
@PersistenceContext(type = PersistenceContextType.EXTENDED, properties = {@PersistenceProperty(name="consistency.level", value="QUORUM")})
```

Yes it works.

```
Also, is there any way to set consistency level for entitymanager rather than directly getting thrift client then I have to execute queries using thrift client not entity manager rite ?
```

use em.setProperty() or create entityManagerFactory with a map prepoulated with such properties. Query execution is via em only, irrespective of whether set via client.setConsistencyLevel or em.setProperty().

```
What is the default consistency level for cassandra using kundera ? Same as default cassandra which is ONE ?
```

Yes it is ONE

-Vivek

**Comment by ansarrafique** [comment]
Yes, but kundera allows you to use both clients thrift and pelops underneath. The example you mentioned only works for Thrift client, what about pelops ? I think em.setProperty() would be the idea choice then what is the name of the property and what is the value ?

**Comment by mevivs** [comment]
setConsistencyLevel is independent of client implementation would work in same way for Thrift,Pelops and DSClient. Same is true with other ways to set consistency level

-Vivek

**Comment by impetus-opensource-admin** [comment]
Closing. Released with 2.11.
